### PR TITLE
Rewriting the military armament loot groups to make them more consistent

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Balance "Changing military armament loot groups so they are more consistent"

#### Purpose of change

Everyone know this classic story, when you loot some military outpost, open the locker, and find a bunch 9x19mm rounds, few STANAG magazines, and one SCAR-H with 3 round inside. It feels really awkward for me, like some guy in green mask put random stuff from his pocket right in this locker, mixing the stuff in the most bizzare combination. I decided to change that

#### Describe the solution

The most perfect solution will require a lot of mapgen changes, to separate weapon and ammo for most places when armament is not designed to be deployed, so this one is the initial part of this changes, aimed on loot groups.
In this pr, for each weapon caliber and type will be maded a separate loot group, that will contain the gun itself, some of his mags, some related ammo (properly packed in special ammunition can), and some misc stuff - some explosives like grenades or C4, or tools like pipe cleaner, or medicine, or attachments for weapons, or some pouches, or some another stuff like night/infrared vision goggles, two way radios or binoculars. In short the idea is SUS_milspec.

#### TODO

- [x] 9mm group: m9, m11, m17, m18
- [x] 9mm submachine gun group: bt_apc9k, hk_mp5, hk_mp5k
- [x] .223 group: m4_carbine, m27_assault_rifle
- [x] .223 group: m4_carbine, m27_assault_rifle
- [x] .223 machine gun group: m249
- [x] .300blk group: m4_carbine, m27_assault_rifle, retool_ar15_300blk
- [x] .300 sniper group: m2010
- [x] .308 machine gun group: m240
- [x] .308 sniper group: m110a1, m14ebr, m24, scar_h
- [x] .50 machine gun group: m2browning
- [x] .50 sniper group: m107a1
- [x] 12 gauge group: m1014, mossberg_590, mossberg_500, m26_mass_standalone
- [x] Grenade 40x46mm group: m320, m79, mgl
- [x] Grenade 40x53mm group: mark19 
- [x] 84x246mm group: m3_carlgustav, m4_carlgustav, AT4
- [x] atgm group: atgm_launcher
- [x] 66mm group: LAW_Packed, m202_flash
- [x] Armor locker
- [x] loot group with only ammo for each beforementioned caliber
- [x] loot group with only weapon and mags for each beforementioned caliber
- [x] replace the old loot groups inside different military maps to new one

#### Existing problems:
- Clips cannot load more than 10 round in a mag, saying "You don't have any 5.56 to reload your magazine". Kevin said its something easy to fix, so i decided to make it anyway
- Stipper clip guides are not exist (because they cannot be used, can be easily added, when the game will be able to use them properly)
- Different ammo boxes has bad naming and description (or do not have at all). Marked as "Lorem ipsum" now
- #57981
- ~~I dont know what to do with some heavy armament, like Mark19, m320 and stuff in same level, that do not exist yet - M202 FLASH, AT4, M72 LAW, M224 mortar etc. They can be here, as "ready to use", but i dont know is making a group for each of them is good idea, or its better to make a one group named like "heavy_armament" and put them in this group with their ammunition (thinking about it make me feel it would be the same amount of work)~~ Made heavy armament group, but will be separated for better tweaking in the future
- ~~Unknown (for me) state of uncommon for our world guns and cartriges - 5x50, 8x40, H&K 12mm etc. Am i need to make groups for them, or i need to ignore them as not existing and highly experimental things~~ do not exist in common military facilities, only lab stuff
- Same with energy weapon
- Ammo. I dont know what another type of ammo is used in military, the biggest part of what i see is common FMJ and, sometimes, its a tracers mixed with FMJ inside the ammo belt

#### Testing

The game works, the loot is pretty oky

#### Screenshots

223, rifles
image

308, sniper rifles
image

223, machine guns
image

9mm, pistols
image

#### Additional context

I want to thank the cleanammocans.com for providing a huge amount of information for this pr 😂

### CODE REVIEW

**Comment by Maleclypse** [comment]
H&K 12mm is almost definitely lab or prototype testing only.

**Comment by lbnesquik** [comment]
I don't get exactly why you are doing it, but that's a big change, and i applaud that much effort.

File: data/json/items/armor/bandolier.json [hunk 1]
```diff
@@ -315,5 +315,103 @@
     "use_action": { "type": "holster", "holster_prompt": "Stash grenades", "holster_msg": "You stash your %s." },
     "flags": [ "WATER_FRIENDLY", "BELTED" ],
     "armor": [ { "coverage": 25, "covers": [ "torso" ], "specifically_covers": [ "torso_upper" ] } ]
+  },
+  {
+    "id": "cheap_ammo_pouch",
+    "type": "ARMOR",
+    "name": "cheap ammo bandolier",
+    "description": "Small bandolier with pouches, designed to store a bunch of ammo clips or hold a whole mag.  This has 4 small pouches, that perfectly fit default STANAG 30-round magazine",
```
Comment (TheShadowFerret) [inline data/json/items/armor/bandolier.json:323]:
```suggestion
    "description": "Small bandolier with pouches, designed to store a bunch of ammo clips or hold a whole mag.  This has four small pouches, that perfectly fit standard STANAG 30-round magazines.",
```

File: data/json/items/armor/bandolier.json [hunk 2]
```diff
@@ -315,5 +315,103 @@
...
+      }
+    ],
+    "flags": [ "BELTED", "OVERSIZE", "WATER_FRIENDLY" ],
+    "armor": [ { "encumbrance": 7, "coverage": 100, "covers": [ "torso" ], "specifically_covers": [ "torso_waist" ] } ]
+  },
+  {
+    "id": "cheap_ammo_pouch_belt308",
+    "type": "ARMOR",
+    "name": "cheap ammo bandolier",
+    "description": "Small bandolier with pouches, designed to store a bunch of ammo clips or hold a whole mag.  This has 4 ",
```
Comment (TheShadowFerret) [inline data/json/items/armor/bandolier.json:394]:
```suggestion
    "description": "Small bandolier with four pouches, designed to store a bunch of ammo clips or hold a whole mag.",
```

File: data/json/items/containers.json [hunk 3]
```diff
@@ -738,6 +738,16 @@
     "symbol": ")",
     "color": "brown"
   },
+  {
+    "id": "box_compact_wood",
+    "type": "GENERIC",
+    "name": { "str": "compact wooden box", "str_pl": "compact wooden boxes" },
+    "copy-from": "ammunition_can_30",
```
Comment (TheShadowFerret) [inline data/json/items/containers.json:745]:
```suggestion
    "copy-from": "ammunition_can_30",
    "material": [ "wood" ],
```
No steel & aluminum wooden boxes for you. 😛

File: data/json/items/containers.json [hunk 4]
```diff
@@ -2791,5 +2801,189 @@
         "max_contains_weight": "500 g"
       }
     ]
+  },
+  {
+    "id": "ammunition_can_30",
+    "type": "GENERIC",
+    "category": "container",
+    "name": "M19A1 ammo can",
```
Comment (TheShadowFerret) [inline data/json/items/containers.json:2809]: These specific names are not useful to the average player, can they use the variant system to give them the toggled generic name?

(I always use the specific names tho <3)

*Applies also to M2A2, PA108 SAW, PA-120 ammo cans + M289 container

File: data/json/items/containers.json [hunk 5]
```diff
@@ -2791,5 +2801,189 @@
...
+        "max_contains_volume": "550 ml",
+        "max_contains_weight": "5000 g"
+      }
+    ],
+    "material": [ "steel" ],
+    "symbol": "0",
+    "color": "black"
+  },
+  {
+    "id": "ammunition_box_308",
```
Comment (TheShadowFerret) [inline data/json/items/containers.json:2922]: I don't think we need this many individually defined ammo boxes, esp. when we have need of a large number of potential ammo boxes that should be defined given how many calibers are represented ingame that should spawn inside them(down with dropping a large pile of loose cartridges!).

File: data/json/items/containers.json (hunk 4 above)
Comment (GuardianDll) [reply data/json/items/containers.json:2809]: Well i dont really know what generic name they can have, their non official names is in id - 30 cal, 50 cal, 50 fat and 40mm. If you have some suggestion I'll be glad to use it, we can put the official names in the description (or just remove it at all, usually people dont know their official names)

File: data/json/items/containers.json (hunk 5 above)
Comment (GuardianDll) [reply data/json/items/containers.json:2922]: Not sure i understand what you mean.

File: data/json/items/containers.json (hunk 4 above)
Comment (TheShadowFerret) [reply data/json/items/containers.json:2810]: I'd go for nonofficial name("7.62x51mm ammo can") w/ official in description.

Comment (GuardianDll) [reply data/json/items/containers.json:2810]: Thats a bad idea, they are reused in many ways, same irl and in the game. `ammunition_can_30` is used to store 9mm, 300blk and 300, when `ammunition_can_50` contain 223, 308 and 50.

File: data/json/items/containers.json (hunk 5 above)
Comment (bombasticSlacks) [reply data/json/items/containers.json:2923]: I think this is a reasonable solution for military. 

In the future paper boxes should also be a thing though but it's a bunch of work to get an automated solution working.

Comment (GuardianDll) [reply data/json/items/containers.json:2923]: but what about the amount of rounds? the box for 20 rounds is not the same as box for 50 rounds, and 20 223 round box is not the same as 20 308 round box. what number we need to pick for rounds to not make 5+ different boxes for each round (which is a huge ton of a boxes that no one want to make, use, and hanlde, obviously)
i use 100x9mm, 30x223, 20x300, 20x12 and 10x50 round boxes for my purpose, but i came to this numbers using ammo cans standarts, so they can be unperfect for some purposes, i think

Comment (bombasticSlacks) [reply data/json/items/containers.json:2923]: what I'm saying is what you are doing and have added is fine for now IMO.

**Comment by GuardianDll** [comment]
40mm grenades (both 40x46 and 40x53) are stored in amount of 32 units and perfectly fit the case with some soft pad on sides
In game you can contain ~110 of them inside. Am i need to left can, ugh, on 2/3 empty, or there is some way to tweak it that i cannot see?

**Comment by TheShadowFerret** [comment]
>40mm grenades (both 40x46 and 40x53) are stored in amount of 32 units

I can only find references in this to the *belt* being 32 grenades long, what are you using for what *a box* can fit 🤔
I re-did their volumes from #40455 & didn't see an issue there.

**Comment by GuardianDll** [comment]
Mk19 (and mk47, cause they are compatible) can with belts are the only info about containing grenade shells that i can find, so i use them as a reference. I thought that even with belt linkage there is too much space left, but officer_dirtbag convince me its the inventory system imperfectness

**Comment by catdach** [comment]
>Stipper clip guides are not exist (because they cannot be used, can be easily added, when the game will be able to use them properly)

*Sigh* Well...there is the `speedloader chute` which is gunmod that is _supposed to_ allow some shotguns to use the 8 round and 6 round shogun-speedloader items, but it's been broken for quite a while. About 6 months ago, I opened some issue requests with some potential solutions. If you want to open up that can of worms, this one might have useful information: #53542. This one is a mess of misunderstandings and miscommunication in the comments that was basically shot down with the mass removal of `clip_size`, still it might be helpful _somehow_ so ill drop it here #53375. I could go on, but I don't want to bog down this PR with this tangential stuff.

**Comment by GuardianDll** [comment]
Think i need only change the names/description and add the loot to maps, and the pr will be ready

File: data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
+[
+  {
+    "type": "item_group",
+    "id": "arsenal_armor_collection",
+    "subtype": "collection",
+    "items": [ { "group": "arsenal_armor_armor", "count": [ 1, 2 ] }, { "group": "arsenal_armor_plates", "count": [ 1, 3 ] } ]
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json:6]:
```suggestion
    "items": [ { "group": "arsenal_armor_armor", "count": [ 1, 2 ] }, { "group": "pristine_esapi_plates", "count": [ 1, 3 ] } ]
```

File: data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  {
+    "type": "item_group",
+    "id": "arsenal_armor_armor",
+    "subtype": "collection",
+    "items": [
+      { "item": "ballistic_vest_esapi", "prob": 100, "count": [ 1, 2 ] },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json:13]:
```suggestion
      { "group": "military_ballistic_vest_pristine", "prob": 100, "count": [ 1, 4 ] },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      { "item": "ballistic_vest_heavy", "prob": 25, "count": [ 1, 2 ] },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json:14]:
```suggestion

```

File: data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+      { "item": "helmet_army", "prob": 100, "count": [ 1, 2 ] },
+      { "item": "tac_helmet", "prob": 50, "count": [ 1, 2 ] },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json:16]:
```suggestion

```

File: data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      { "item": "tac_fullhelmet", "prob": 25, "count": [ 1, 2 ] },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json:17]:
```suggestion
      { "item": "tac_fullhelmet", "prob": 25, "count": [ 1, 2 ] },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    "items": [ { "group": "arsenal_armor_armor", "count": [ 1, 2 ] }, { "group": "arsenal_armor_plates", "count": [ 1, 3 ] } ]
+  },
+  {
+    "type": "item_group",
+    "id": "arsenal_armor_armor",
+    "subtype": "collection",
+    "items": [
+      { "item": "ballistic_vest_esapi", "prob": 100, "count": [ 1, 2 ] },
+      { "item": "ballistic_vest_heavy", "prob": 25, "count": [ 1, 2 ] },
+      { "item": "helmet_army", "prob": 100, "count": [ 1, 2 ] },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/arsenal/armor.json:15]:
```suggestion
      { "item": "helmet_army", "prob": 100, "count": [ 1, 4 ] },
```

**Review by bombasticSlacks — COMMENTED** [review]
this gets rid of the cop stuff and makes it so you get an assortment of vests and only military grade plates as spares. 

if you don't have them elsewhere shooters earmuffs should maybe be included in the armor collection.

**Pull request merged**